gp-includes/things/original.php

```diff
@@ -107,8 +107,15 @@ public function import_for_project( $project, $translations ) {
 
 		$possibly_added = $possibly_dropped = array();
 
-		foreach( $translations->entries as $entry ) {
+		foreach( $translations->entries as $key => $entry ) {
 			$wpdb->queries = array();
+
+			// Context needs to match VARCHAR(255) in the database schema.
+			if ( gp_strlen( $entry->context ) > 255 ) {
+				$entry->context = gp_substr( $entry->context, 0, 255 );
+				$translations->entries[ $entry->key() ] = $entry;
```

Comment:
Thought about this a bit and I think can't remove the other entry or stop the processing because otherwise an existing original doesn't get updated.